### PR TITLE
Updating WireGuard Appliance and its Dependencies

### PR DESCRIPTION
**Pull request by Christie018**
New update code attempts to fix lighttpd service issue and landing page. Added working code from Jeremy D v17-update to improve lading page. Appliance has been updated and tested to ensure all services working correctly. Readme.rst file has been reviewed and updated to ensure all links are valid and relevant.

### CODE REVIEW

File: conf.d/main [hunk 1]
```diff
@@ -9,7 +9,12 @@ apt install --assume-yes wireguard
 # webserver related
 mkdir -p /var/www/wireguard/logs
 chown www-data:www-data /var/www/wireguard/logs
-lighty-disable-mod ssl
+
+service lighttpd restart
```
Comment (JedMeister) [inline conf.d/main]: This line should be required. I'm guessing that you added it for debugging/testing and just forgot to remove it?

File: conf.d/main [hunk 2]
```diff
@@ -9,7 +9,12 @@ apt install --assume-yes wireguard
...
+lighty-enable-mod wireguard
+service lighttpd force-reload
```
Comment (JedMeister) [inline conf.d/main]: Same with these 3 lines.

As it doesn't actually use the webserver during build time (like some other appliances do; e.g. for curl install of a PHP app). As such, strictly speaking, you don't even need to start lighty. Having said that, perhaps starting then stopping it is a good idea to test that the config is correct?! Although if that was why you're doing this, you could just use the command to test lighty config (in the conf script). I.e. `lighttpd -tt -f /etc/lighttpd/lighttpd.conf`

File: conf.d/main [hunk 3]
```diff
@@ -19,3 +24,4 @@ rm -rf /var/www/{js,css,images}
 
 # ensure ipv4 forwarding is enabled
 sed -i 's/#* *net.ipv4.ip_forward=.*/net.ipv4.ip_forward=1/g' /etc/sysctl.conf
+
```
Comment (JedMeister) [inline conf.d/main]: There is no need for a trailing white space (i.e. a blank line at the end) here. It's not a big deal and I probably wouldn't have even noticed it - except you added it. 

For some reason many of the old scripts (before my time) have this, but there is no good reason for it. So please feel free to remove them from any files you work on.

**Review by JedMeister — COMMENTED** [review]
Looks great Mattie. Just a couple of notes re the conf script.

**Comment by JedMeister** [comment]
Also:
> Removed bug notice from setup.rst since I was prompted for server/client config on firstboot.

Good catch and great thinking. However, a couple of things:

1. this is the [relevant bug](https://github.com/turnkeylinux/tracker/issues/1646). When you fix a bug, please note that in the changelog (including the bug number - e.g. something like `Fixed XYZ - closes #9999`) and in the PR description (with a link to the bug instead of just the text `#9999`).
2. having said that, looking at the issue, it appears to be related to the TurnKey Hub. I just logged into the Hub it looks like it's definitely primarily a bug within the Hub (which we'll need to get Alon to address - FYI it's occurred because I didn't give him the adequate info at the time). Looking at the firstboot inithook, we may need to tweak it some more, but until Alon applies the fix to the Hub, the warning should probably stay there (FYI, when launching from the Hub, you are presented with a form that you fill out. The values from your input are then used by the inithooks - so when launching from the Hub, you don't need to manually go through them on firstboot).

So to summarise, I think that the warning needs to stay for now sorry...

File: conf.d/main (hunk 1 above)
Comment (Christie018) [reply conf.d/main]: Did you mean to say these lines should not be required? I can replace these lines with the one you suggested to test lighttpd capabilities during conf.d build.

File: conf.d/main (hunk 2 above)
Comment (Christie018) [reply conf.d/main]: The force reload did come up on screen in the logs while building the appliance, I don't believe it stopped the build but it did say to run `service lighttpd force reload` after running the `lighty-enable-mod wireguard` . After the build, I also re ran both `lighty-enable-mod wireguard` and `lighty-enable-mod ssl` but the terminal replied they were already enabled.

File: conf.d/main (hunk 3 above)
Comment (Christie018) [reply conf.d/main]: I removed this from the code per PR review comment.

File: conf.d/main (hunk 2 above)
Comment (Christie018) [reply conf.d/main]: I went through and added the test script per PR review comment. I also went through and removed the lines restarting/reloading the lighttpd service in conf.d/main and retested appliance.

**Review by OnGle — APPROVED** [review]
lgtm :)

**Comment by JedMeister** [comment]
I'm going to merge this for now as this PR is great.

Unfortunately though, it's not ready to build as I'll need to discuss the outstanding issue (https://github.com/turnkeylinux/tracker/issues/1646) with Alon. We may need to make changes to the inithooks to get it all working as required.
